### PR TITLE
Feature/use latest io ts

### PR DESCRIPTION
**Pull request by twonamejames**
This change upgrades the versions of io-ts and fp-ts to version 2. This is to fix type errors caused by incompatible versions of fp-ts and io-ts in a project I'm working on that has dependencies on version 2 of both packages as well as this package.

I also took the opportunity to update any outdated packages that could be updated without breaking.

Luckily only a few things broke when upgrading io-ts to version 2. I only fixed errors reported by tsc. I don't know if other issues still exist, but all the tests seem to be passing.

Because I believe this is a breaking change, I bumped the major version of this package as well.

### CODE REVIEW

**Comment by Ahryman40k** [comment]
Hi @twonamejames, 
Thank you for the update. Indeed it was a job I need to do. 
I agree with everything except the library version., It will be nice if it can stay on version 4. 
All your changes doesn't break FHIR objects so it's still R4. 
I did myself a mistake with FHIR version and I'm waiting next release to fix it. Indeed, my lib should have been in version 4.0.0.x then in version 4.0.1.x  but I missed a '0'. 
Can you just increase the patch number ?

Thank for you contribution and your time :)

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@ahryman40k/ts-fhir-types",
-  "version": "4.0.29",
```
Comment (Ahryman40k) [inline package.json:3]: Because library is FHIR R4 standard, it will be nice to maintain 4.0.x format.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+  "version": "5.0.0",
```
Comment (Ahryman40k) [inline package.json]:
```suggestion
  "version": "4.0.30",
```

**Review by Ahryman40k — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (twonamejames) [reply package.json:3]: No problem. I committed the suggested changes, so now it stays at 4.0.x

**Review by Ahryman40k — APPROVED** [review]

**Comment by Ahryman40k** [comment]
Thank you very much for this !

**Pull request merged**